src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -559,7 +555,7 @@ private static void AddProjectReferences(PackageSpec spec, IEnumerable<IMSBuildI
 
         private static Tuple<List<string>, ProjectRestoreReference> GetProjectRestoreReference(IMSBuildItem item)
         {
-            var frameworks = GetFrameworks(item).ToList();
+            var frameworks = GetFrameworks(item).AsList();
```

Comment:
This returns a hashset, so as List won't do anything. 

This change is unnecessary. There's no functional change here.
LINQ generates enumerates which had to be converted to List